cmd/juju/commands/upgradejuju.go

```diff
@@ -363,7 +369,8 @@ func (c *upgradeJujuCommand) Run(ctx *cmd.Context) (err error) {
 				return block.ProcessBlockedError(err, block.BlockChange)
 			}
 		}
-		if err := client.SetModelAgentVersion(context.chosen); err != nil {
```

Comment:
Any specific reasons why you did not want to use BestFacadeVersion check to determine which call to make?